// @toverux/ngsweetalert2/dist/types+es2015-modules/src/swal.component.js
import { Component, EventEmitter, Inject, Input, Output } from '@angular/core';
import swal from 'sweetalert2';
import { SWAL_DEFAULTS } from './di';
var SwalComponent = (function () {
    function SwalComponent(defaultSwalOptions) {
        this.defaultSwalOptions = defaultSwalOptions;
        this.confirm = new EventEmitter();
        this.cancel = new EventEmitter();
        this.show = this.show.bind(this);
    }
    SwalComponent.prototype.show = function () {
        var _this = this;
        var options = Object.assign({
            type: this.type,
            title: this.title,
            text: this.text,
            html: this.html,
        }, this.defaultSwalOptions, this.options);
        var promise = swal(options);
        promise.then(function (success) { return _this.confirm.emit(success); }, function (dismiss) { return _this.cancel.emit(dismiss); });
        return promise;
    };
    return SwalComponent;
}());
export { SwalComponent };
SwalComponent.decorators = [
    { type: Component, args: [{
                selector: 'swal',
                template: ''
            },] },
];
/** @nocollapse */
SwalComponent.ctorParameters = function () { return [
    { type: undefined, decorators: [{ type: Inject, args: [SWAL_DEFAULTS,] },] },
]; };
SwalComponent.propDecorators = {
    'type': [{ type: Input },],
    'title': [{ type: Input },],
    'text': [{ type: Input },],
    'html': [{ type: Input },],
    'options': [{ type: Input },],
    'confirm': [{ type: Output },],
    'cancel': [{ type: Output },],
};
