// @toverux/ngsweetalert2/__ivy_ngcc__/dist/types+es2015-modules/src/swal.directive.js
import { Directive, EventEmitter, HostListener, Inject, Input, Output } from '@angular/core';
import swal from 'sweetalert2';
import { SWAL_DEFAULTS } from './di';
import * as ɵngcc0 from '@angular/core';
var SwalDirective = (function () {
    function SwalDirective(defaultSwalOptions) {
        this.defaultSwalOptions = defaultSwalOptions;
        this.confirm = new EventEmitter();
        this.cancel = new EventEmitter();
        this.modalOptions = {};
    }
    Object.defineProperty(SwalDirective.prototype, "swal", {
        set: function (options) {
            if (Array.isArray(options)) {
                this.modalOptions.title = options[0], this.modalOptions.text = options[1];
                if (options.length >= 3) {
                    this.modalOptions.type = options[2];
                }
            }
            else {
                this.modalOptions = options;
            }
        },
        enumerable: true,
        configurable: true
    });
    SwalDirective.prototype.onHostClicked = function (event) {
        var _this = this;
        event.preventDefault();
        event.stopImmediatePropagation();
        var options = Object.assign({}, this.defaultSwalOptions, this.modalOptions);
        swal(options).then(function (success) { return _this.confirm.emit(success); }, function (dismiss) { return _this.cancel.emit(dismiss); });
    };
SwalDirective.ɵfac = function SwalDirective_Factory(t) { return new (t || SwalDirective)(ɵngcc0.ɵɵdirectiveInject(SWAL_DEFAULTS)); };
SwalDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SwalDirective, selectors: [["", "swal", ""]], hostBindings: function SwalDirective_HostBindings(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵlistener("click", function SwalDirective_click_HostBindingHandler($event) { return ctx.onHostClicked($event); });
    } }, inputs: { swal: "swal" }, outputs: { confirm: "confirm", cancel: "cancel" } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SwalDirective, [{
        type: Directive,
        args: [{ selector: '[swal]' }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [SWAL_DEFAULTS]
            }] }]; }, { confirm: [{
            type: Output
        }], cancel: [{
            type: Output
        }], swal: [{
            type: Input
        }], onHostClicked: [{
            type: HostListener,
            args: ['click', ['$event']]
        }] }); })();
    return SwalDirective;
}());
export { SwalDirective };
/** @nocollapse */
SwalDirective.ctorParameters = function () { return [
    { type: undefined, decorators: [{ type: Inject, args: [SWAL_DEFAULTS,] },] },
]; };
SwalDirective.propDecorators = {
    'swal': [{ type: Input },],
    'confirm': [{ type: Output },],
    'cancel': [{ type: Output },],
    'onHostClicked': [{ type: HostListener, args: ['click', ['$event'],] },],
};
