// @toverux/ngsweetalert2/__ivy_ngcc__/dist/types+es2015-modules/src/swal.component.js
import { Component, EventEmitter, Inject, Input, Output } from '@angular/core';
import swal from 'sweetalert2';
import { SWAL_DEFAULTS } from './di';
import * as ɵngcc0 from '@angular/core';
var SwalComponent = (function () {
    function SwalComponent(defaultSwalOptions) {
        this.defaultSwalOptions = defaultSwalOptions;
        this.confirm = new EventEmitter();
        this.cancel = new EventEmitter();
        this.show = this.show.bind(this);
    }
    SwalComponent.prototype.show = function () {
        var _this = this;
        var options = Object.assign({
            type: this.type,
            title: this.title,
            text: this.text,
            html: this.html,
        }, this.defaultSwalOptions, this.options);
        var promise = swal(options);
        promise.then(function (success) { return _this.confirm.emit(success); }, function (dismiss) { return _this.cancel.emit(dismiss); });
        return promise;
    };
SwalComponent.ɵfac = function SwalComponent_Factory(t) { return new (t || SwalComponent)(ɵngcc0.ɵɵdirectiveInject(SWAL_DEFAULTS)); };
SwalComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SwalComponent, selectors: [["swal"]], inputs: { type: "type", title: "title", text: "text", html: "html", options: "options" }, outputs: { confirm: "confirm", cancel: "cancel" }, decls: 0, vars: 0, template: function SwalComponent_Template(rf, ctx) { }, encapsulation: 2 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SwalComponent, [{
        type: Component,
        args: [{
                selector: 'swal',
                template: ''
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [SWAL_DEFAULTS]
            }] }]; }, { confirm: [{
            type: Output
        }], cancel: [{
            type: Output
        }], type: [{
            type: Input
        }], title: [{
            type: Input
        }], text: [{
            type: Input
        }], html: [{
            type: Input
        }], options: [{
            type: Input
        }] }); })();
    return SwalComponent;
}());
export { SwalComponent };
/** @nocollapse */
SwalComponent.ctorParameters = function () { return [
    { type: undefined, decorators: [{ type: Inject, args: [SWAL_DEFAULTS,] },] },
]; };
SwalComponent.propDecorators = {
    'type': [{ type: Input },],
    'title': [{ type: Input },],
    'text': [{ type: Input },],
    'html': [{ type: Input },],
    'options': [{ type: Input },],
    'confirm': [{ type: Output },],
    'cancel': [{ type: Output },],
};
