// @toverux/ngsweetalert2/__ivy_ngcc__/dist/types+es2015-modules/index.js
import { NgModule } from '@angular/core';
import { swalDefaultsProvider } from './src/di';
import { SwalComponent } from './src/swal.component';
import { SwalDirective } from './src/swal.directive';
import * as ɵngcc0 from '@angular/core';
export * from './src/swal.component';
export * from './src/swal.directive';
var SweetAlert2Module = (function () {
    function SweetAlert2Module() {
    }
    SweetAlert2Module.forRoot = function (defaultSwalOptions) {
        return {
            ngModule: SweetAlert2Module,
            providers: [swalDefaultsProvider(defaultSwalOptions)]
        };
    };
SweetAlert2Module.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: SweetAlert2Module });
SweetAlert2Module.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function SweetAlert2Module_Factory(t) { return new (t || SweetAlert2Module)(); }, providers: [swalDefaultsProvider()] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(SweetAlert2Module, { declarations: function () { return [SwalComponent, SwalDirective]; }, exports: function () { return [SwalComponent, SwalDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SweetAlert2Module, [{
        type: NgModule,
        args: [{
                declarations: [SwalComponent, SwalDirective],
                exports: [SwalComponent, SwalDirective],
                providers: [swalDefaultsProvider()]
            }]
    }], function () { return []; }, null); })();
    return SweetAlert2Module;
}());
export { SweetAlert2Module };
/** @nocollapse */
SweetAlert2Module.ctorParameters = function () { return []; };
