// angular2-ladda/__ivy_ngcc__/module/module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var ladda_directive_1 = require("./ladda.directive");
var ladda_config_1 = require("./ladda-config");
var ɵngcc0 = require('@angular/core');
var ɵngcc1 = require('./ladda.directive');
var LaddaModule = (function () {
    function LaddaModule() {
    }
    LaddaModule.forRoot = function (config) {
        return {
            ngModule: LaddaModule,
            providers: [
                { provide: ladda_config_1.LaddaConfig, useValue: config }
            ]
        };
    };
LaddaModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: LaddaModule });
LaddaModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function LaddaModule_Factory(t) { return new (t || LaddaModule)(); } });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(LaddaModule, { declarations: [ɵngcc1.LaddaDirective], exports: [ɵngcc1.LaddaDirective] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(LaddaModule, [{
        type: core_1.NgModule,
        args: [{
                declarations: [ladda_directive_1.LaddaDirective],
                exports: [ladda_directive_1.LaddaDirective]
            }]
    }], function () { return []; }, null); })();
    return LaddaModule;
}());
/** @nocollapse */
LaddaModule.ctorParameters = function () { return []; };
exports.LaddaModule = LaddaModule;

exports.LaddaDirective = ɵngcc1.LaddaDirective;
